differ/renderer.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright © 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differ
+
+import (
+	"bytes"
+	"encoding/json"
+	httputils "github.com/RedHatInsights/insights-operator-utils/http"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/RedHatInsights/ccx-notification-service/conf"
+	"github.com/RedHatInsights/ccx-notification-service/types"
+	utypes "github.com/RedHatInsights/insights-results-types"
+	"github.com/rs/zerolog/log"
+)
+
+func renderReportsForCluster(
+	config conf.DependenciesConfiguration,
+	clusterName types.ClusterName,
+	reports []types.ReportItem,
+	ruleContent types.RulesMap) ([]types.RenderedReport, error) {
+
+	rendererURL := httputils.SetHTTPPrefix(config.TemplateRendererServer + config.TemplateRendererEndpoint)
```

Comment:
I'd replace `config.TemplateRendererServer + config.TemplateRendererEndpoint` with `url.JoinPath(config.TemplateRendererServer, config.TemplateRendererEndpoint)` in order to avoid errors because of missing "/"

Reply:
looks like https://pkg.go.dev/net/url#JoinPath was added in go1.19...

Reply:
Yes, I changed my local version of Go because of that :D Should I change it in `go.mod`?

Reply:
I don't think we can. The image we run our apps in is go1.16. I would rather find another workaround for this matter instead of having to change our base images :grimacing:

Reply:
Ok, I replaced it with a workaround found [here](https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join), even though it is a bit messy :)

Reply:
Ok, I replaced it with a workaround found [here](https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join), even though it is a bit messy :)

Reply:
@JiriPapousek what are you trying to solve?

Reply:
Just the fact that `config.TemplateRendererServer + config.TemplateRendererEndpoint` could break the functionality e.g. when `TemplateRendererServer` ends with '/' and `TemplateRendererEndpoint` starts with it.

Reply:
maybe `strings.Join([]string{url, endpoint}, "/")` is easier

Reply:
@JiriPapousek make it as new configuration item and you are done :)

Reply:
@tisnik By new configuration item you mean putting serverURL and endpointURL into one string?

Reply:
@tisnik By new configuration item you mean putting serverURL and endpointURL into one string?

Reply:
Ok, should be fixed by moving URL joining to configuration setup.